step2/hackatom-stash-controller.sh

```diff
@@ -1,79 +1,106 @@
+echo
+echo "Create an authz authorization to perform MsgUndelegate."
+$BUILD_CMD tx authz grant $CONTROLLER_ADDRESS unbond --from $STASH --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block --fees $MIN_FEES -y --allowed-validators $VALIDATOR_ADDRESS
 
-# Query authorization grants granted by STASH
+echo
+echo "Create an authz authorization to perform MsgRedelegate."
+$BUILD_CMD tx authz grant $CONTROLLER_ADDRESS redelegate --from $STASH --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block --fees $MIN_FEES -y --allowed-validators $VALIDATOR_ADDRESS
+
+echo
+echo "Query authorization grants granted by STASH"
 $BUILD_CMD q authz granter-grants $STASH_ADDRESS --chain-id $CHAIN_ID
 
-# Query all the authz grants authorized to CONTROLLER from STASH
+echo
+echo "Query all the authz grants authorized to CONTROLLER from STASH"
 $BUILD_CMD q authz grants $STASH_ADDRESS $CONTROLLER_ADDRESS --chain-id $CHAIN_ID
 
-# Set up a feegrant from STASH to CONTROLLER
-$BUILD_CMD tx feegrant grant $STASH $CONTROLLER_ADDRESS --spend-limit 10000stake --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block -y
+echo
+echo "Set up a feegrant of 200stake from STASH to CONTROLLER to pay tx fees"
+$BUILD_CMD tx feegrant grant $STASH $CONTROLLER_ADDRESS --spend-limit 200stake --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block --fees $MIN_FEES -y
 
-# Query the feegrant from STASH to CONTROLLER 
+echo
+echo "Query the feegrant from STASH to CONTROLLER"
 $BUILD_CMD q feegrant grant $STASH_ADDRESS $CONTROLLER_ADDRESS --chain-id $CHAIN_ID
 
-# Query all the grants for CONTROLLER
+echo
+echo "Query all the grants for CONTROLLER"
 $BUILD_CMD q feegrant grants $CONTROLLER_ADDRESS --chain-id $CHAIN_ID
 
-# Delegate liquid coins to validator from STASH
-$BUILD_CMD tx staking delegate $VALIDATOR_ADDRESS 10stake --from $STASH --keyring-backend test --chain-id $CHAIN_ID --generate-only> tx.json 
+echo
+echo "Delegate liquid coins (42stake) to validator from STASH (no signature from STASH needed!)"
+$BUILD_CMD tx staking delegate $VALIDATOR_ADDRESS 42stake --from $STASH --keyring-backend test --chain-id $CHAIN_ID --generate-only> tx.json 
 
-# Query the delegations from the STASH account
-$BUILD_CMD q staking delegations $STASH_ADDRESS --chain-id $CHAIN_ID
+echo
+echo "Exec the MsgDelegate from CONTROLLER on behalf of STASH key (STASH pays for fees!)"
+$BUILD_CMD tx authz exec tx.json --from $CONTROLLER --fee-account $STASH_ADDRESS --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block --fees $MIN_FEES -y
 
-# Exec the MsgDelegate from CONTROLLER on behalf of STASH key
-$BUILD_CMD tx authz exec tx.json --from $CONTROLLER --keyring-backend test --chain-id $CHAIN_ID --broadcast-mode block -y
+echo
+echo "Query again the feegrant from STASH to CONTROLLER (it's less than the initial 200stake)"
+$BUILD_CMD q feegrant grant $STASH_ADDRESS $CONTROLLER_ADDRESS --chain-id $CHAIN_ID
```

Comment:
I added this new query.

Initially, the feegrant is 200stake. After one tx, here, it drops to 180stake.